bin/jruby-truffle

```diff
@@ -3,4 +3,32 @@
 bin="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
 root="$(dirname "$bin")"
 
-exec "$root"/tool/jt.rb ruby --no-print-cmd "$@"
+set -e
+
+function mtime {
+    uname_str=$(uname)
+    if [ "$uname_str" = 'Linux' ]; then
+        stat -c '%Y' $1 2> /dev/null || echo 0
+    elif [ "$uname_str" = 'Darwin' ]; then
+        stat -f '%m' $1 2> /dev/null || echo 0
+    else
+        echo "unknown platform $uname_str"
+        exit 1
+    fi
+}
+
+if [ -n "${RUBY_BIN}" ]; then
+    ${RUBY_BIN}
+else
+    mx_jar="$root"/mxbuild/dists/ruby.jar
+    mvn_jar="$root"/lib/jruby-truffle.jar
+    mx_time=$(mtime ${mx_jar})
+    mvn_time=$(mtime ${mvn_jar})
+
+    if [ ${mx_time} -gt ${mvn_time} ]; then
+        exec "$root"/tool/jruby_mx "$@"
+    else
+        exec "$root"/bin/jruby -X+T "$@"
+    fi
+fi
```

Comment:
Is it a problem currently? Otherwise we'll get rid of the Maven build soon, so this will get much simpler.